src/dmd/root/filename.d

```diff
@@ -889,4 +893,77 @@ version(Windows)
 
         return F(ret);
     }
+
+}
+
+version(Posix)
+{
+
+    /**************************
+     * Takes a callable F and applies it to the result of converting
+     * `fileName` to an absolute file path (char*)
+     */
+    auto absPathThen(alias F)(const(char)* fileName)
+    {
+        char[1024] realPathBuffer;
+        auto absPath = realpath(fileName, &realPathBuffer[0]);
+        return F(absPath);
+    }
+
+    extern(C) char* realpath(const(char)* file_name, char* resolved_name) @nogc nothrow;
```

Comment:
1. Why not import this from druntime?
2. This should be private

Reply:
1. That didn't occur to me. I didn't even know importing from druntime was allowed in dmd.
2. Agreed.